importer/Bosnia.py

```diff
@@ -0,0 +1,321 @@
+        Since the mapping file covers all that's in the
+        source data, there's not trying / reporting to do.
+        """
+        municip_dict = self.data_files["_static"]["bosnia_adm"]
+        municip_raw = self.municipality.strip()
+        municip_match = utils.get_item_from_dict_by_key(dict_name=municip_dict,
+                                                        search_term=municip_raw,
+                                                        search_in="itemLabel")
+        self.add_statement("located_adm", municip_match[0], refs=self.source)
+
+    def set_street_address(self):
+        addr_raw = self.address.strip()
+        if addr_raw != "N/A" and not addr_raw.startswith("http"):
+            # yes this is needed
+            self.add_statement("located_street", addr_raw, refs=self.source)
+
+    def update_labels(self):
+        labels = {
+            "en": self.english_name,
+            "bs": self.short_name
+        }
+        for lang in labels:
+            self.add_label(lang, labels[lang])
+        self.add_alias("bs", self.official_name)  # 2 bs labels
+
+    def update_descriptions(self):
+        """
+        Set description in bs.
+
+        Use object_type, decapitalized.
+        """
+        bs_desc = self.object_type[0].lower() + self.object_type[1:]
+        self.add_description("bs", bs_desc)
+
+    def set_coords(self):
+        """
+        Set coordinates.
+
+        Try to parse the non-split text value...
+        """
+        coords_raw = self.coordinates.replace("°", "")
+        if "N/A" not in coords_raw and "x" not in coords_raw:
+            if "N" in coords_raw and "E" in coords_raw:
+                split_parts = coords_raw.split("N")
+                try:
+                    coord_tuple = (float(split_parts[0].strip()),
+                                   float(split_parts[1].strip()[:-1].strip()))
+                    self.add_statement(
+                        "coordinates", coord_tuple, refs=self.source)
+                except ValueError:
+                    print("Nope coords: {}".format(self.coordinates))
+
+    def construct_wd_item(self, mapping, data_files=None):
+        """
+        Create the empty structure of the data object.
+
+        Like Monument.py but simplify mapping loading.
+        """
+        self.wd_item = {}
+        self.wd_item["upload"] = True
+        self.wd_item["statements"] = {}
+        self.wd_item["labels"] = {}
+        self.wd_item["aliases"] = {}
+        self.wd_item["descriptions"] = {}
+        self.wd_item["disambiguators"] = {}
+        self.wd_item["wd-item"] = None
+        self.mapping = mapping
+        self.create_source()
+        self.set_country()
+        self.set_adm_location()
+        self.set_coords()
+        self.set_is()
+        self.set_heritage()
+        self.set_described()
+        self.set_street_address()
+        self.update_descriptions()
+        self.update_labels()
+
+
+def generate_filenames(tablename, timestamp):
```

Comment:
why not just `import importer` in the start then call `importer.make_filenames` instead?

Reply:
True for several of the other functions as well and would make it easier to see which parts of the code are new and which are re-used.

Reply:
Because at that point I had no way of knowing whether I'd be able to use the importer functions or whether and to what degree they'd have to be modified.